### PR TITLE
Minor changes, game fully playable now

### PR DESCRIPTION
**Pull request by wickyakloe**
@ignatiusukwuoma

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,12 +1,49 @@
...
+                        <p>ON</p>
+                    </div>
+                    <div id="off-button">
+                        <button id="reset" class="button shadow light"></button>
+                        <p>OFF</p>
+                    </div>
+                </div>
+                <div id="strictMode">
+                        <button id="strictBut" class="shadow light" value="OFF"><span id="strictOnOff">STRICT MODE: OFF</span></button>
+                    </div>
```
Comment (ignatiusukwuoma) [inline index.html]: properly indent this line the and one above

File: static/js/game.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+// fill Playlist
+function fillPlaylist () {
+  playlist.push(genRandomInt())
+  showLevel()
+  if (showLevel() === '21') {
+    victory()
+    $('#reset').mousedown()
+    $('#reset').mouseup()
```
Comment (ignatiusukwuoma) [inline static/js/game.js]: what does this code and the one above do?

File: static/js/game.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+  } else {
+    console.log('Awaiting victory')
```
Comment (ignatiusukwuoma) [inline static/js/game.js]: Remove all console.logs when you're ready to submit

File: static/js/game.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+// yellow = 3
+function clickButton (color, value) {
+  changeColor(color)
+  checkValue(value)
+  userInput.push(value)
+  return userInput
+}
+
+// Test userInput array against playlist array
+function tester (cb1) {
```
Comment (ignatiusukwuoma) [inline static/js/game.js]: cb1 as a variable name is not easy to understand. Use userInput instead, or some other variable name that is easily comprehensible.

File: static/js/game.js [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+      } else {
+        console.log('values equal')
+      }
+    })
+  } else {
+    if (cb1[cb1.length - 1] !== playlist[playlist.length - 1]) {
+      if ($('#strictBut').val() === 'OFF') {
+        console.log('values are not equal2')
+        userInput = []
+        playButtons()
```
Comment (ignatiusukwuoma) [inline static/js/game.js]: Do something to show user that they made a wrong attempt before calling playButtons(). I would expect an error sign on `#level` and an error sound.

File: static/js/game.js [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+function genRandomInt () {
+  var genNo = Math.floor(Math.random() * Math.floor(4))
+  return genNo
+}
+
+// fill Playlist
+function fillPlaylist () {
+  playlist.push(genRandomInt())
+  showLevel()
+  if (showLevel() === '21') {
```
Comment (ignatiusukwuoma) [inline static/js/game.js]: we're calling showLevel() twice, once in the previous line and the second here. What about calling it once in the previous line and saving the return value in a variable.

File: static/js/game.js [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+          userInput = []
+          $('#level').text('00')
+          fillPlaylist()
+          playButtons()
+        }
+      } else {
+        console.log('values equal')
+      }
+    })
+  } else {
```
Comment (ignatiusukwuoma) [inline static/js/game.js]: We're doing almost the same thing in the if-part of this conditional statement as in the else part. Can you refactor this?

File: static/js/game.js [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+    victory()
+    $('#reset').mousedown()
+    $('#reset').mouseup()
+  } else {
+    console.log('Awaiting victory')
+  }
+}
+
+function addRemClick () {
+  if ($('#red')[0].hasAttribute('onclick') && $('#blue')[0].hasAttribute('onclick') &&
```
Comment (ignatiusukwuoma) [inline static/js/game.js]: Why do you need the [0]s?
Also, I think we can do this in a loop.

File: static/js/game.js [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+  $('#green')[0].hasAttribute('onclick') && $('#yellow')[0].hasAttribute('onclick')) {
+    $('#red').removeAttr('onclick')
+    $('#green').removeAttr('onclick')
+    $('#blue').removeAttr('onclick')
+    $('#yellow').removeAttr('onclick')
+  } else {
+    $('#red').attr('onclick', 'tester(clickButton("red", 0 ))')
+    $('#green').attr('onclick', 'tester(clickButton("green", 1 ))')
+    $('#blue').attr('onclick', 'tester(clickButton("blue", 2 ))')
+    $('#yellow').attr('onclick', 'tester(clickButton("yellow", 3 ))')
```
Comment (ignatiusukwuoma) [inline static/js/game.js]: We should abstract this else part into a function

File: index.html (hunk 1 above)
Comment (wickyakloe) [reply index.html]:
```html
                <div id="strictMode">
                    <button id="strictBut" class="shadow light" value="OFF"><span id="strictOnOff">STRICT MODE: OFF</span></button>
                </div>
```

File: static/js/game.js (hunk 2 above)
Comment (wickyakloe) [reply static/js/game.js]: Added function victoryCheck()

```javascript
function fillPlaylist () {
  playlist.push(genRandomInt())
  showLevel()
  victoryCheck()
}

function victoryCheck () {
  if (showLevel() === '21') {
    victory()
    $('#reset').mousedown()
    $('#reset').mouseup()
  } else {
    console.log('Awaiting victory')
  }
}
```

File: static/js/game.js (hunk 4 above)
Comment (wickyakloe) [reply static/js/game.js]: Updated:
- replaced cb1 with userinput
- updated code from else { if ( ) { } } to else if ( ) { }
```javascript
// Test userInput array against playlist array
function tester (userinput) {
  if (userinput.length !== playlist.length) {
    userinput.forEach(function (number, index) {
      if (userinput[index] !== playlist[index]) {
        // Check strictmode on or off
        if ($('#strictBut').val() === 'OFF') {
          console.log('values are not equal2')
          userInput = []
          playButtons()
        } else {
          playlist = []
          userInput = []
          $('#level').text('00')
          fillPlaylist()
          playButtons()
        }
      } else {
        console.log('values equal')
      }
    })
  } else if (userinput[userinput.length - 1] !== playlist[playlist.length - 1]) {
    if ($('#strictBut').val() === 'OFF') {
      console.log('values are not equal2')
      userInput = []
      playButtons()
    } else {
      playlist = []
      userInput = []
      $('#level').text('00')
      fillPlaylist()
      playButtons()
    }
  } else {
    console.log('values equal2')
    fillPlaylist()
    playButtons()
    userInput = []
  }
}
```

**Pull request closed without merging**